### ISSUE TITLE
npm does not upgrade package to the latest version

### ISSUE THREAD

**Issue by ziyaddin** [npm/npm#14570]
#### I'm opening this issue because:

  - [x] npm is doing something I don't understand.

#### What's going wrong?

I make changes to my own `index.js` file in my `dev` package and publish it with a new version to the registry and then I run command to upgrade it in my machine globally, but it does not upgrade it to the latest version, as it shows "Wanted" version is less than "Latest" version.

#### How can the CLI team reproduce the problem?

```bash
$ npm outdated -g
Package    Current  Wanted  Latest  Location
test-zizi    1.4.0   1.4.0   1.5.0
```

My package.json file:

```json
{
  "name": "test-zizi",
  "version": "1.5.0",
  "description": "just a test for zizi",
  "files": [
    "index.js"
  ],
  "bin": "./index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [
    "test"
  ],
  "author": "",
  "license": "ISC"
}

```

### supporting information:

 - `npm -v` prints: 3.10.9
 - `node -v` prints: v5.12.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Linux

**Comment by KenanY** [comment]
@ziyaddin So this still occurs after running `npm install -g test-zizi@1.5.0`?

**Comment by ziyaddin** [comment]
@KenanY no. It is being solved on this way. What is the issue that prevents it "Wanted" to be the "Lates" version?

**Comment by KenanY** [comment]
@ziyaddin Hmm, what command specifically do you mean by this (just asking to make sure)?:

> run command to upgrade it in my machine globally

**Comment by ziyaddin** [comment]
`npm upgrade -g test-zizi`

**Comment by legodude17** [comment]
`npm update` is a bit of a foot gun. A better way to do that is `npm install -g test-zizi@latest`.

**Comment by pdelre** [comment]
`npm install -g npm@latest` also fixed an issue where I'd see npm getting updated to a Pre-Release tag when using `npm update -g`.

For example:
```
$ npm -g outdated
Package  Current  Wanted  Latest  Location
npm        4.4.2   4.4.2   4.4.1
```

![image](https://cloud.githubusercontent.com/assets/1379248/23947920/1dc4bcea-0957-11e7-8de0-3656f7fa44a3.png)

**Comment by ziyaddin** [comment]
@pdelre it still shows 4.4.1 as the latest version for me.

**Comment by pdelre** [comment]
Yes, it shows that 4.4.1 as the latest, but when running `npm -g update npm` it downloads 4.4.2. running `npm -g install npm@latest` installs 4.4.1. I'm not a node/npm developer, but it would seem to me that it hits github's api for releases and doesn't filter out "Pre-release".

**Comment by legodude17** [comment]
That is not what is happening. `npm -g install npm@latest` installs the release with the `latest` tag, which is `4.4.1`, but `npm update -g npm` uses the last version published, which is `next`, or `4.4.2`.

**Comment by Manoj002** [comment]
@ziyaddin @KenanY @legodude17 @pdelre 
Hello guys,
im facing this issue now as im updating my npm but everytime it shows me the version of 6.0.0 while i updated it to 6.0.1

Any help will be highly appreciated!

**Comment by ziyaddin** [comment]
Hello @Manoj002,
Have you tried `sudo npm install -g npm@latest` ?

**Comment by Manoj002** [comment]
Hello @ziyaddin , thank you for your suggestion mate

i tried your suggested command as the package got updated to 6.0.1 and again after typing npm -v,
it shows 6.0.0

**Comment by SamuelBrucksch** [comment]
Hi,

we are having similiar issues. When we release a fix for a lower version, then the latest tag is set to that build (its build infrastructure specific, we can not change much here). Unfortunately things like 2.x or * do not work anymore, as they seem to depend on the latest tag instead of checking the semver version.

Example:
2.1 is released -> latest is 2.1 after release
2.2 is released -> latest is 2.2 after release
2.1.1 is released -> latest is 2.1.1

so if in the end i say install 2.x it installs 2.1.1

Is there a way to avoid this behaviour? I thought npm makes use of semver versioning and should be able to see, that 2.2 is a higher version than 2.1.1

Cheers

Sam
